### PR TITLE
Refactor get_element_effect_info() to avoid sprintf().  Does part of …

### PR DESCRIPTION
**Pull request by backwardsEric**
…the work of resolving https://github.com/hengband/hengband/issues/2858 and https://github.com/hengband/hengband/issues/2859 .

### CODE REVIEW

File: src/mind/mind-elementalist.cpp [hunk 1]
```diff
@@ -409,62 +409,46 @@ static mind_type get_elemental_info(PlayerType *player_ptr, int spell_idx)
  * @brief 元素魔法呪文の効果表示文字列を取得
  * @param player_ptr プレイヤー情報への参照ポインタ
  * @param spell_idx 呪文番号
- * @param p バッファ
- * @return なし(pを更新)
+ * @return std::string スペルに関する印刷可能なコメント付き
```
Comment (habu1010) [inline src/mind/mind-elementalist.cpp:412]: The natural Japanese expression is "魔法の効果を表す文字列".

Comment (backwardsEric) [reply src/mind/mind-elementalist.cpp]: The forced push should have changed the machine-translation in the comment to your suggestion.

File: src/mind/mind-elementalist.cpp [hunk 2]
```diff
@@ -800,7 +783,7 @@ static bool get_element_power(PlayerType *player_ptr, SPELL_IDX *sn, bool only_b
 
                     PERCENTAGE chance = decide_element_chance(player_ptr, spell);
                     int mana_cost = decide_element_mana_cost(player_ptr, spell);
-                    get_element_effect_info(player_ptr, i, comment);
+                    auto comment = get_element_effect_info(player_ptr, i);
```
Comment (Hourier) [inline src/mind/mind-elementalist.cpp:786]: Variables other than primitive types which don't change are desirable to be declared with const.
Would you please add it?
(This is the same for other PRs.)

Comment (backwardsEric) [reply src/mind/mind-elementalist.cpp]: The latest forced push adds const to the declaration at mind/mind-elementalist.cpp:786.

**Review by Hourier — APPROVED** [review]
Thank you for reflecting our comments.
I checked the current codes and found no problem.
I approve this.

**Pull request merged**